cpp/arcticdb/storage/mongo/mongo_storage.cpp

```diff
@@ -35,7 +35,9 @@ void MongoStorage::do_write(Composite<KeySegmentPair>&& kvs) {
     (fg::from(kvs.as_range()) | fg::move | fg::groupBy(fmt_db)).foreach([&](auto &&group) {
         for (auto &kv : group.values()) {
             auto collection = collection_name(kv.key_type());
-            client_->write_segment(db_, collection, std::move(kv));
+            std::string potential_error_msg = fmt::format("Mongo error while putting key {}", kv.key_view());
+            auto success = client_->write_segment(db_, collection, std::move(kv));
+            util::check(success, potential_error_msg);
```

Comment:
The reason why I build error message first is that `kv` is being moved with `std::move`. Using `kv` after `std::move` is [undefined behaviour](https://www.sites.google.com/a/chromium.org/dev/rvalue-references?tmpl=%2Fsystem%2Fapp%2Ftemplates%2Fprint%2F&showPrintDialog=1#:~:text=However..-,5.%20You%20must%20not%20use%20a%20variable%20after%20calling%20std%3A%3Amove()%20on%20it,-.).

Reply:
Makes sense, still it would be cheaper to copy the `kv.key_view()` before moving than building the `potential_error_msg`.

Anyway, this doesn't matter as it surely isn't the bottleneck. Happy to leave as is.

Reply:
Fixed in [68222fb](https://github.com/man-group/ArcticDB/pull/1383/commits/68222fb3803137c1849edc977467cb2390462eab)